outlines/models/mlxlm.py

```diff
@@ -0,0 +1,240 @@
+import dataclasses
+from typing import TYPE_CHECKING, Generator, Iterator, List, Optional, Tuple, Union
+
+from .transformers import TransformerTokenizer
+
+if TYPE_CHECKING:
+    import mlx.nn as nn
+    from transformers import PreTrainedTokenizer
+
+    from outlines.generate.api import GenerationParameters, SamplingParameters
+    from outlines.processors import BaseLogitsProcessor
+
+try:
```

Comment:
Does that mean the  user must have `mlx` installed, whether they want to use this integration or not?

Reply:
It will attempt to import, but the module will load fine if `mlx` isn't installed because the exception passes.

Reply:
I think it’s cleaner to import the libraries directly in the methods/functions where they’re used.

Reply:
Fixed!